ui/src/app/pages/data/criteria-search.tsx

```diff
@@ -126,31 +127,30 @@ export class CriteriaSearch extends React.Component<Props, State>  {
       mode = 'tree';
     }
     this.setState({backMode, hierarchyNode, mode});
-    this.subscription = currentCohortCriteriaStore.subscribe(currentCohortCriteria => {
-      if (source === 'criteria') {
+    if (source === 'criteria') {
+      this.subscription = currentCohortCriteriaStore.subscribe(currentCohortCriteria => {
         this.setState({selectedCriteriaList: currentCohortCriteria});
-      }
-    });
-    this.subscription.add(currentConceptStore.subscribe(currentConcepts => {
-      if (source === 'concept') {
-        this.setState({selectedCriteriaList: currentConcepts});
-      }
-    }));
+      });
+    } else {
+      this.subscription = currentConceptStore.subscribe(currentConcepts => {
```

Comment:
I'm getting `currentConcepts` as undefined here when viewing a saved concept set, so there's an empty space where the table should be. Not sure why but seems like it doesn't get the updated store value when the concept set is loaded and set in `concept-set-details.tsx`.
<img width="1294" alt="Screen Shot 2020-10-22 at 2 13 13 PM" src="https://user-images.githubusercontent.com/40036095/96919064-d9ecce80-1470-11eb-9029-f1a5cd7c112e.png">

Reply:
Sorry, I got my stores mixed up and my last comment doesn't make sense. We need to set `currentConceptStore` for all domains in `concept-set-details.tsx`. Currently, it only gets set for surveys.

Reply:
Done and also i enclosed it within a flag check that i had forgotten to do before